From jakj@j-a-k-j.com Wed Apr  4 04:39:32 2007
From: "John Anthony Kazos Jr." <jakj@j-a-k-j.com>
Date: Wed, 4 Apr 2007 07:39:17 -0400 (EDT)
Subject: Kobject: kobject_uevent.c: Collapse unnecessary loop nesting (top_kobj)
To: gregkh@suse.de
Cc: linux-kernel@vger.kernel.org
Message-ID: <alpine.DEB.0.83.0704040730530.25382@sigma.j-a-k-j.com>

From: John Anthony Kazos Jr. <jakj@j-a-k-j.com>

Collapses a do..while() loop within an if() to a simple while() loop for 
simplicity and readability.

Signed-off-by: John Anthony Kazos Jr. <jakj@j-a-k-j.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject_uevent.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

--- a/lib/kobject_uevent.c
+++ b/lib/kobject_uevent.c
@@ -95,10 +95,8 @@ int kobject_uevent_env(struct kobject *k
 
 	/* search the kset we belong to */
 	top_kobj = kobj;
-	if (!top_kobj->kset && top_kobj->parent) {
-		do {
-			top_kobj = top_kobj->parent;
-		} while (!top_kobj->kset && top_kobj->parent);
+	while (!top_kobj->kset && top_kobj->parent) {
+		top_kobj = top_kobj->parent;
 	}
 	if (!top_kobj->kset) {
 		pr_debug("kobject attempted to send uevent without kset!\n");
